packages/task/src/browser/quick-open-task.ts

```diff
@@ -322,6 +307,31 @@ export class QuickOpenTask implements QuickOpenModel, QuickOpenHandler {
         });
     }
 
+    private addTaskItemsByTaskScope(groupedTasks: Map<string, TaskConfiguration[]>, token: number, isMulti: boolean, isFirstGroup: boolean, taskScopeString: string): boolean {
```

Comment:
I had some offline discussions with @alvsan09 and we will try to think of a cleaner solution. 
At the moment I'm hesitant with the method since:
- returning a `boolean` seems counterintuitive to the method name of adding tasks (`false` makes it seem like an error would occur).
- documentation would certainly be helpful.
- we can try to refactor to not have duplication with `global`, `workspace` and `folder` scopes.

Reply:
The new commit refactors this function and its calling block, so there is no longer a returned boolean and the function is now reused.
Thanks Vince !